kv/client.go

```diff
@@ -76,8 +76,14 @@ func (cfg *Config) RegisterFlagsWithPrefix(flagsPrefix, defaultPrefix string, f
 	if flagsPrefix == "" {
 		flagsPrefix = "ring."
 	}
+
+	// Allow clients to override default store by setting it before calling this method.
+	if cfg.Store == "" {
```

Comment:
if `cfg.Store != ""` should it's value be validated? to ensure it matches one from the list of valid values?

Reply:
I think we can assume that programmer supplies correct value. In any case, validation happens later when client is created: https://github.com/grafana/dskit/blob/c40fe15dd22eabcb69afa06edc1eae53d20e6c07/kv/client.go#L132

Reply:
fair enough :+1: